### PR TITLE
parseschema2.py Java code generation

### PR DESCRIPTION
**Pull request by AFFogarty**
Added the ability for parseschema2.py to generate Java code.

You can't actually run the code right now, as pymei has yet to be ported to Java.  But if pymei is eventually ported to Java, then we would already have Java code generation for an MEI schema.

### CODE REVIEW

File: tools/langs/java.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        match = re.match(regmatch, line)
+        if match:
+            if match.group("elementName") in methods.keys():
+                contents[i] = methods[match.group("elementName")].lstrip("\n") + "\n"
+
+    f = open(os.path.join(directory, codefile), 'w')
+    f.writelines(contents)
+    f.close()
+
+def capitalize_first_letter(text):
```
Comment (alastair) [inline tools/langs/java.py]: https://docs.python.org/2/library/stdtypes.html#str.capitalize ?

Comment (AFFogarty) [inline tools/langs/java.py]: That function lowercases all subsequent chars, though. 
 
My function preserves CamelCase.

Comment (alastair) [inline tools/langs/java.py]: Ah. Carry on

**Comment by AFFogarty** [comment]
I have no idea why Travis would fail.

**Pull request merged**